### PR TITLE
refactor: add get recent collections endpoint

### LINKED ISSUE #392: Add GET recent collections

### PR DESCRIPTION
**Pull request by pyphilia**
close #392

### CODE REVIEW

File: src/services/item/plugins/published/service.ts [hunk 1]
```diff
@@ -134,4 +134,14 @@ export class ItemPublishedService {
 
     return filterOutHiddenItems(repositories, items);
   }
+
+  // filter out by categories, not defined will return all items
```
Comment (pyphilia) [inline src/services/item/plugins/published/service.ts:138]: remove this comment.

File: src/services/item/plugins/published/test/index.test.ts [hunk 2]
```diff
@@ -185,6 +185,54 @@ describe('Item Published', () => {
...
+      let collections: Item[];
+      let categories;
+
+      beforeEach(async () => {
+        ({ app } = await build({ member: null }));
+        member = await saveMember(BOB);
+        collections = await saveCollections(member);
+        categories = await saveCategories();
+
+        // add category to a non-published item
```
Comment (pyphilia) [inline src/services/item/plugins/published/test/index.test.ts:200]: remove category related variables for "recent" tests.

File: src/services/item/plugins/published/test/index.test.ts [hunk 3]
```diff
@@ -185,6 +185,54 @@ describe('Item Published', () => {
...
+
+      it('Get 2 most recent collections', async () => {
+        const res = await app.inject({
+          method: HttpMethod.GET,
+          url: `${ITEMS_ROUTE_PREFIX}/collections/recent`,
+          query: { limit: 2 },
+        });
+        expect(res.statusCode).toBe(StatusCodes.OK);
+
+        const result = collections.sort(({ createdAt: a }, { createdAt: b }) => (a > b ? -1 : 1));
```
Comment (morganridel) [inline src/services/item/plugins/published/test/index.test.ts:213]: I'm partial to `collections.sort((a, b) => b.createdAt - a.createdAt);` for sorting usually, but it's personal :)

**Review by morganridel — APPROVED** [review]
LGTM

**Review by dialexo — APPROVED** [review]
LGTM 🍭 see attached question

File: src/services/item/plugins/published/index.ts [hunk 4]
```diff
@@ -89,5 +90,16 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       });
     },
   );
+
+  fastify.get<{ Querystring: { limit?: number } }>(
+    '/collections/recent',
+    {
+      preHandler: fastify.fetchMemberInSession,
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/services/item/plugins/published/index.ts:97]: exactly!

**Pull request merged**